webapp/src/client.js

```diff
@@ -15,29 +15,36 @@ export default class Client {
         return this.doPost(`${this.url}/settings`, meeting);
     }
 
-    doGet = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
-        headers['X-Timezone-Offset'] = new Date().getTimezoneOffset();
+    doGet = async (url, headers = {}) => {
+        return this.doFetch(url, { headers });
+    }
+
+    doPost = async (url, body, headers = {}) => {
+        return this.doFetch(url, {
+            method : 'POST',
+            body: JSON.stringify(body),
+            headers: {...headers, ...{
+                'Content-Type' : 'application/json'
+            }}
+        });
+    }
 
-        const response = await request.
-            get(url).
-            set(headers).
-            accept('application/json');
+    doFetch = async (url, { method = 'GET', body = null, headers = {} }) => {
+        const response = await fetch(url, {
+            method,
+            body,
+            headers: {...this.getCommonHeaders(), ...headers}
+        });
```

Comment:
We should include all options that `Client4.getOptions` provides, instead of just the headers. Also since this function is so small, and it's the only function calling `this.getCommonHeaders()`, I suggest removing `this.getCommonHeaders()` and inlining those headers here instead:
```suggestion
		const options = Client4.getOptions({
		    method,
		    body,
		    headers: {
		    	...headers
		    	'X-Timezone-Offset': new Date().getTimezoneOffset(),
		    	Accept: 'application/json',
		    },
        });
        const response = await fetch(url, options);
```